### PR TITLE
Enable FindBugs and fix violations

### LINKED ISSUE #14: Enable FindBugs plugin and fix any violations

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #14

### CODE REVIEW

**Comment by meltsufin** [comment]
@loosebazooka PTAL

File: pom.xml [hunk 1]
```diff
@@ -47,35 +48,36 @@
...
-      <version>3.3.9</version>
-    </dependency>
+
     <dependency>
       <groupId>com.google.cloud.tools</groupId>
       <artifactId>app-tools-lib-for-java</artifactId>
       <version>0.1.0-SNAPSHOT</version>
     </dependency>
+
     <dependency>
```
Comment (loosebazooka) [inline pom.xml]: what is commons.io used for that guava can't do?

Comment (meltsufin) [inline pom.xml]: Guava doesn't provide a method to delete a directory recursively. It used to, but the new versions don't.

Comment (loosebazooka) [inline pom.xml]: Oh I see, to clean the staging directory.

File: src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java [hunk 2]
```diff
@@ -47,20 +46,20 @@ public void outputLine(String line) {
     }
   };
 
-  @Component
+  @Parameter(defaultValue = "${pluginDescriptor}", readonly = true)
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java]: Why is this a parameter?

File: pom.xml (hunk 1 above)
Comment (meltsufin) [inline pom.xml]: Yep. Are you doing this in Gradle too?

File: src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java (hunk 2 above)
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java]: It's the same thing. `@Component ` is now deprecated. Hence the change. `readonly=true` makes it invisible to users of the plugin.

File: src/main/java/com/google/cloud/tools/maven/StageMojo.java [hunk 3]
```diff
@@ -169,10 +169,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
       try {
         FileUtils.deleteDirectory(stagingDirectory);
       } catch (IOException e) {
-        new MojoFailureException("Unable to delete staging directory.", e);
+        throw new MojoFailureException("Unable to delete staging directory.", e);
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/maven/StageMojo.java]: A bug that findbugs actually found!

File: pom.xml (hunk 1 above)
Comment (loosebazooka) [inline pom.xml]: gradle implements delete themselves

**Comment by loosebazooka** [comment]
:ship:

**Pull request merged**